src/Game.cs

```diff
@@ -854,18 +865,38 @@ private void RunLoop()
 
 			while (RunApplication)
 			{
-				FNAPlatform.PollEvents(
-					this,
-					ref currentAdapter,
-					textInputControlDown,
-					textInputControlRepeat,
-					ref textInputSuppress
-				);
 				Tick();
 			}
 			OnExiting(this, EventArgs.Empty);
 		}
 
+		private TimeSpan AdvanceElapsedTime()
+		{
+			long currentTicks = gameTimer.Elapsed.Ticks;
+			TimeSpan timeAdvanced = TimeSpan.FromTicks(currentTicks - previousTicks);
+			accumulatedElapsedTime += timeAdvanced;
+			previousTicks = currentTicks;
+			return timeAdvanced;
+		}
+
+		/* To calculate the sleep precision of the OS, we take the worst case
+		 * time spent sleeping over the results of previous requests to sleep 1ms.
+		 */
+		private void UpdateEstimatedSleepPrecision(TimeSpan timeSpentSleeping)
+		{
+			previousSleepTimes[sleepTimeIndex] = timeSpentSleeping;
```

Comment:
You know the previous worst case - it's saved in worstCaseSleepPrecision. You also know your current index. So the only way the worst case changes is if you either 1) just got a new worst case, or 2) the worst case was the oldest entry on the list. You can check that quickly with something like this:

```
if (timeSpentSleeping >= worstCaseSleepPrecision)
{
    worstCaseSleepPrecision = timeSpentSleeping;
}
else if (previousSleepTimes[sleepTimeIndex] == worstCaseSleepTime)
{
    var maxSleepTime = TimeSpan.MinValue;
    for (int i = 0; i < previousSleepTimes.Count; i++)
    {
        if (previousSleepTimes[i] > maxSleepTime)
        {
            maxSleepTime = sleepTime;
        }
    }
    worstCaseSleepPrecision = maxSleepTime;
}

previousSleepTimes[sleepTimeIndex] = timeSpentSleeping;
sleepTimeIndex = (sleepTimeIndex + 1) & SLEEP_TIME_MASK;
```

Reply:
Crap, obvious catch, thanks for finding this >_<

Reply:
Seconded, thanks for this.